kubernetes-client/src/main/java/io/micronaut/kubernetes/client/ModelMapper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client;
+
+import io.kubernetes.client.apimachinery.GroupVersionKind;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.util.Strings;
+import org.apache.commons.lang3.tuple.MutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is inspired by the implementation of the <a href="https://github.com/kubernetes-client/java/blob/release-13/util/src/main/java/io/kubernetes/client/util/ModelMapper.java">io.kubernetes.client.util.ModelMapper.java</a>.
+ * <p>
+ * The difference is that this class doesn't prefetch all model classes but instead does the resolution
+ * on demand. Also the integration with {@link io.kubernetes.client.Discovery} is placed in the {@link DiscoveryCache}
+ * bean.
+ *
+ * @author Pavol Gressa
+ * @since 3.1
+ */
+public class ModelMapper {
```

Comment:
Note: Another handy class that resolves the resource version from the resource class name